includes/snmp.inc.php

```diff
@@ -37,22 +37,63 @@ function prep_snmp_setting($device, $setting)
     }
 }//end prep_snmp_setting()
 
+/**
+ * @param $device
+ * @return array $extra will contain a list of mib dirs
+ */
+function get_mib_dir($device)
+{
+    global $config;
+    $extra = array();
+
+    if (file_exists($config['mib_dir'] . '/' . $device['os'])) {
+        $extra[] = $config['mib_dir'] . '/' . $device['os'];
+    }
+
+    if (file_exists($config['mib_dir'] . '/' . $device['os_group'])) {
+        $extra[] = $config['mib_dir'] . '/' . $device['os_group'];
+    }
+
+    if (isset($config['os_groups'][$device['os_group']]['mib_dir'])) {
+        if (is_array($config['os_groups'][$device['os_group']]['mib_dir'])) {
+            foreach ($config['os_groups'][$device['os_group']]['mib_dir'] as $k => $dir) {
+                $extra[] = $config['mib_dir'] . '/' . $dir;
+            }
+        }
+    }
+
+    if (isset($config['os'][$device['os']]['mib_dir'])) {
```

Comment:
Not sure about this one either.  Our snmp commands allow mib directories to be specified still, so a custom mibdir can be specified for each call.